### PR TITLE
Solve Tikas' issue with promises

### PR DESCRIPTION
**Pull request by SEQUOIIA**
Story details: https://app.clubhouse.io/devex/story/1999

### CODE REVIEW

**Comment by kim-lindhard-dfds** [comment]
[//]: # (4744e426-5795-11e9-8647-d663bd873d93)

|| **CodeScene Delta Analysis Results** |
| --- | ---- |
| **Risk** |10|
| **Quality Gates** |OK|
| **Description** |The change is high risk and adds 117 lines, deletes 32 lines of code in 8 files. The risk is lower since it is a very experienced author with many contributions.|
| **Warnings** |Delta Analysis Info<br/><ul><li>2 files in this change set did not exist with the same name on the main branch. This is probably because those files have been renamed or introduced in the commits under analysis. Skipping code health checks for those files.</li></ul>|

File: server/src/server/api/middleware/requestErrorMiddleware.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+) => {
+    console.error(err);
+
+    response.setHeader('Content-Type', 'application/json');
+
+    switch (err.name.valueOf()) {
+        case new TopicAlreadyExistsException().name.valueOf():
+            response.status(409).json({errName: err.name, errMessage: err.message});
+            break;
+        case new ServiceAccountAlreadyExistsException().name.valueOf():
```
Comment (kim-lindhard-dfds) [inline server/src/server/api/middleware/requestErrorMiddleware.ts:19]: [NP] you can combine the 
```
  case new TopicAlreadyExistsException().name.valueOf():
 case new ServiceAccountAlreadyExistsException().name.valueOf():
```

cases

File: server/src/server/api/middleware/requestLoggerMiddleware.ts [hunk 2]
```diff
@@ -6,7 +6,8 @@ export const RequestLogger = (
   next: NextFunction
 ) => {
 
-  console.info(`${request.method} ${request.path}`);
+  // 4/2/2020, 10:39:06 AM: [1347a8a782381] Request GET - /
+  console.info(`${new Date().toLocaleString()}: [${(request as any).id}] Request ${request.method} - ${request.path}`);
```
Comment (kim-lindhard-dfds) [inline server/src/server/api/middleware/requestLoggerMiddleware.ts:20]: Lets output this in json. and use a UTC timestamp

File: server/src/server/api/middleware/responseLoggerMiddleware.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+export const ResponseLogger = (
+  request: Request,
+  response: Response,
+  next: NextFunction
+) => {
+
+  // 4/2/2020, 10:39:06 AM: [1347a8a782381] Response GET 200 - /
+  console.info(`${new Date().toLocaleString()}: [${(request as any).id}] Response ${request.method} ${response.statusCode} - ${request.path}`);
+
```
Comment (kim-lindhard-dfds) [inline server/src/server/api/middleware/responseLoggerMiddleware.ts:20]: Lets output this in json. and use a UTC timestamp

Comment (kim-lindhard-dfds) [reply server/src/server/api/middleware/responseLoggerMiddleware.ts:20]: where is the correlationId?

File: server/src/server/api/middleware/requestLoggerMiddleware.ts (hunk 2 above)
Comment (kim-lindhard-dfds) [reply server/src/server/api/middleware/requestLoggerMiddleware.ts:20]: where is the correlationId?

**Review by kim-lindhard-dfds — CHANGES_REQUESTED** [review]

File: server/src/server/api/middleware/requestLoggerMiddleware.ts (hunk 2 above)
Comment (SEQUOIIA) [reply server/src/server/api/middleware/requestLoggerMiddleware.ts:20]: `[${(request as any).id}]` is the correlationId

File: server/src/server/api/middleware/responseLoggerMiddleware.ts (hunk 3 above)
Comment (SEQUOIIA) [reply server/src/server/api/middleware/responseLoggerMiddleware.ts:20]: `[${(request as any).id}]` is the correlationId

File: server/src/server/api/middleware/requestErrorMiddleware.ts (hunk 1 above)
Comment (SEQUOIIA) [reply server/src/server/api/middleware/requestErrorMiddleware.ts:19]: Any specific implementation proposal?

**Comment by kim-lindhard-dfds** [comment]
[//]: # (4744e426-5795-11e9-8647-d663bd873d93)

|| **CodeScene Delta Analysis Results** |
| --- | ---- |
| **Risk** |10|
| **Quality Gates** |OK|
| **Description** |The change is high risk and adds 138 lines, deletes 34 lines of code in 8 files. The risk is lower since it is a very experienced author with many contributions.|
| **Warnings** |Delta Analysis Info<br/><ul><li>2 files in this change set did not exist with the same name on the main branch. This is probably because those files have been renamed or introduced in the commits under analysis. Skipping code health checks for those files.</li></ul>|

File: server/src/server/api/middleware/requestLoggerMiddleware.ts (hunk 2 above)
Comment (SEQUOIIA) [reply server/src/server/api/middleware/requestLoggerMiddleware.ts:20]: Changed output to JSON and now using UTC - 0f5d7ca

File: server/src/server/api/middleware/responseLoggerMiddleware.ts (hunk 3 above)
Comment (SEQUOIIA) [reply server/src/server/api/middleware/responseLoggerMiddleware.ts:20]: Changed output to JSON and now using UTC - 0f5d7ca

**Comment by kim-lindhard-dfds** [comment]
[//]: # (4744e426-5795-11e9-8647-d663bd873d93)

|| **CodeScene Delta Analysis Results** |
| --- | ---- |
| **Risk** |10|
| **Quality Gates** |OK|
| **Description** |The change is high risk and adds 138 lines, deletes 34 lines of code in 8 files. The risk is lower since it is a very experienced author with many contributions.|
| **Warnings** |Delta Analysis Info<br/><ul><li>2 files in this change set did not exist with the same name on the main branch. This is probably because those files have been renamed or introduced in the commits under analysis. Skipping code health checks for those files.</li></ul>|

**Comment by kim-lindhard-dfds** [comment]
[//]: # (4744e426-5795-11e9-8647-d663bd873d93)

|| **CodeScene Delta Analysis Results** |
| --- | ---- |
| **Risk** |10|
| **Quality Gates** |OK|
| **Description** |The change is high risk and adds 138 lines, deletes 39 lines of code in 8 files. The risk is lower since it is a very experienced author with many contributions.|
| **Warnings** |Delta Analysis Info<br/><ul><li>2 files in this change set did not exist with the same name on the main branch. This is probably because those files have been renamed or introduced in the commits under analysis. Skipping code health checks for those files.</li></ul>|

**Comment by kim-lindhard-dfds** [comment]
[//]: # (4744e426-5795-11e9-8647-d663bd873d93)

|| **CodeScene Delta Analysis Results** |
| --- | ---- |
| **Risk** |10|
| **Quality Gates** |OK|
| **Description** |The change is high risk and adds 140 lines, deletes 41 lines of code in 8 files. The risk is lower since it is a very experienced author with many contributions.|
| **Warnings** |Delta Analysis Info<br/><ul><li>2 files in this change set did not exist with the same name on the main branch. This is probably because those files have been renamed or introduced in the commits under analysis. Skipping code health checks for those files.</li></ul>|

**Comment by kim-lindhard-dfds** [comment]
[//]: # (4744e426-5795-11e9-8647-d663bd873d93)

|| **CodeScene Delta Analysis Results** |
| --- | ---- |
| **Risk** |10|
| **Quality Gates** |OK|
| **Description** |The change is high risk and adds 143 lines, deletes 42 lines of code in 8 files. The risk is lower since it is a very experienced author with many contributions.|
| **Warnings** |Delta Analysis Info<br/><ul><li>2 files in this change set did not exist with the same name on the main branch. This is probably because those files have been renamed or introduced in the commits under analysis. Skipping code health checks for those files.</li></ul>|

**Review by kim-lindhard-dfds — APPROVED** [review]
Looks good. Lets merge!

**Pull request merged**